### DISCUSSION TITLE
How to attach requestId to logger?

### DISCUSSION THREAD

**Discussion by nmichkarev** [RobinTail/express-zod-api#718] [Q&A] [answered]
Hi!
I have a question. We need to log an error information in ResultHandler with request id. Is it possible to define logger child for separate request with own metadata(requestId)? Or at least pass this requestId to ResultHandler?

**Comment by RobinTail** [comment]
Hello @nmichkarev ,

You can `.addExpressMiddleware()` with a package `express-request-id` ([check it out](https://www.npmjs.com/package/express-request-id)) or similar. In this case the `request` would have the property `id`.

However, currently you might need also a custom ResultHandler, since the default one is not aware of such additional features, and moreover, the request id does not exist at the moment of logger creation.

In case you have any ideas on how it could be easily customizable, please let me know.

**Comment by RobinTail** [comment]
Regarding child loggers except the ResultHandler, the following approach can be used: https://github.com/RobinTail/express-zod-api/discussions/1465

**Comment by RobinTail** [comment, marked as answer]
The feature of `childLoggerProvider` is now available in recently released version 16.4.0 🚀 
, @nmichkarev

**Comment by RobinTail** [comment]
Update on this topic.
Starting v19.2.0 `.child()` method added to the built-in logger
